SplunkAppForWazuh/appserver/static/js/controllers/overview/mitre-ids/overviewMitreIdsCtrl.js

```diff
@@ -0,0 +1,344 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+define([
+  '../../module',
+  './lib/mitre-techniques',
+  './lib/discover-search-helper',
+  '../../../services/visualizations/table/table',
+  '../../../services/visualizations/inputs/time-picker',
+  'FileSaver',
+], function (app, mitre_techniques, SearchHelper, Table, TimePicker) {
+  'use strict'
+
+  class OverviewMitreIds {
+    /**
+     * Class constructor
+     * @param {Object} $urlTokenModel
+     * @param {Object} $scope
+     * @param {Object} $currentDataService
+     * @param {Object} $state
+     * @param {Object} $notificationService
+     * @param {Object} $requestService
+     * @param {Object} agentData
+     * @param {*} $reportingService
+     */
+
+    constructor(
+      $scope,
+      $currentDataService,
+      $state,
+      $notificationService,
+      $requestService,
+      mitre_tactics,
+      $mdDialog,
+      $dateDiffService,
+      $urlTokenModel,
+      extensions
+    ) {
+      this.scope = $scope
+      this.currentDataService = $currentDataService
+      this.currentDataService.addFilter(
+        `{"rule.mitre.id{}":"*", "implicit":true, "onlyShow":true}`
+      )
+      this.modalOpen = false
+      this.modalInitialized = false
+      this.api = this.currentDataService.getApi()
+      this.apiReq = $requestService.apiReq
+      this.state = $state
+      this.scope.extensions = extensions
+      this.notification = $notificationService
+      this.currentClusterInfo = this.currentDataService.getClusterInfo()
+      this.filters = this.currentDataService.getSerializedFilters(false)
+      this.mitre_tactics = mitre_tactics
+      this.mitre_techniques = mitre_techniques
+      this.$mdDialog = $mdDialog
+      this.urlTokenModel = $urlTokenModel
+      this.setBrowserOffset = $dateDiffService.setBrowserOffset
+      this.reloadFilters = this.reloadFilters.bind(this)
+      this.vizz = []
+
+      try {
+        this.scope.loadingModalData = false
+        this.scope.$applyAsync();
+        // Initialize time tokens to default
+        if (
+          !this.urlTokenModel.has('earliest') &&
+          !this.urlTokenModel.has('latest')
+        ) {
+          this.urlTokenModel.set({ earliest: '0', latest: '' })
+        }
+        this.timePicker = new TimePicker(
+          '#timePicker',
+          this.reloadFilters
+        )
+        if (
+          !this.urlTokenModel.has('form.when.earliest') &&
+          !this.urlTokenModel.has('form.when.latest')
+        ) {
+          this.urlTokenModel.set({ 'form.when.earliest': '0', 'form.when.latest': '' })
+        }
+        if (this.clusterInfo && this.clusterInfo.status === 'enabled') {
+          this.scope.searchBarModel.node_name = nodes || []
+        }
+      } catch (error) { } //eslint-disable-line
```

Comment:
Could this catch at least show the error?

Reply:
Thanks for noticing it!